TypeScript errors in charts/analyze/org/ visualization components (37 type mismatches)

## Description

Multiple TypeScript errors exist in the organization activity chart visualization components under `apps/web/charts/analyze/org/`. These errors indicate type mismatches that could lead to runtime issues.

## Affected Files

- `charts/analyze/org/activity-action-ratio/visualization.ts`
- `charts/analyze/org/activity-action-top-repos/metadata.ts`
- `charts/analyze/org/activity-action-top-repos/visualization.ts`
- `charts/analyze/org/activity-efficiency/visualization.ts`
- `charts/analyze/org/activity-map/metadata.ts`
- `charts/analyze/org/activity-map/visualization.ts`
- `charts/analyze/org/activity-open-to-close/metadata.ts`
- `charts/analyze/org/activity-open-to-close/visualization.ts`
- `charts/analyze/org/activity-open-to-first-response/visualization.ts`
- `charts/analyze/org/company/metadata.ts`
- `charts/analyze/org/company/visualization.tsx`
- `charts/analyze/org/engagement-scatter/metadata.ts`
- `charts/analyze/org/engagement-scatter/visualization.ts`
- `charts/analyze/org/pull-requests-open-to-review/visualization.ts`

## Error Types

1. **Implicit any types**: ECharts formatter callbacks missing type annotations
2. **Property access errors**: Accessing properties that don't exist on typed interfaces (e.g., `TopLevelFormatterParams`)
3. **Index access errors**: Using string/number indices on types that don't support them
4. **Possibly undefined**: Accessing properties on potentially undefined values
5. **Missing type declarations**: `d3-cloud` module lacks TypeScript declarations
6. **Invalid option properties**: Using ECharts options that don't exist in type definitions

## Example Errors

```
charts/analyze/org/activity-action-ratio/visualization.ts(161,9): error TS7053: Element implicitly has an 'any' type
charts/analyze/org/activity-action-ratio/visualization.ts(165,17): error TS2339: Property 'name' does not exist on type 'TopLevelFormatterParams'
charts/analyze/org/company/visualization.tsx(3,19): error TS7016: Could not find a declaration file for module 'd3-cloud'
```

## Suggested Fix

1. Add proper type annotations to all ECharts formatter callbacks
2. Use type guards or optional chaining for potentially undefined values
3. Add `@types/d3-cloud` or create custom type declarations
4. Review ECharts option types and remove invalid properties
5. Consider adding `@ts-expect-error` comments with explanations for intentional type overrides

## Impact

These errors are currently masked by `ignoreBuildErrors` in tsconfig.json, which means they could cause runtime errors in production.

## Related

- #<issue-number> - Critical: ignoreBuildErrors masks TypeScript errors in production builds

--- a/apps/web/charts/analyze/org/activity-action-ratio/visualization.ts
+++ b/apps/web/charts/analyze/org/activity-action-ratio/visualization.ts
@@ -157,12 +157,12 @@ export default function (
       position: function (pos, params, dom, rect, size) {
         // tooltip will be fixed on the right if mouse hovering on the left,
         // and on the left if hovering on the right.
-        var obj = { top: 60 };
+        var obj: Record<string, number> = { top: 60 };
         obj[['left', 'right'][+(pos[0] < size.viewSize[0] / 2)]] = 5;
         return obj;
       },
       formatter: (params) => {
-        const { name, value } = params;
+        const { name, value } = params as any;
         if (activity === 'pull-requests/self-merged' && name === 'self-merged') {
           return `<b>${name}</b>
             <div>${value} PRs(${((value / sum[0]) * 100).toFixed(0)}%)</div>

--- a/apps/web/charts/analyze/org/activity-action-top-repos/metadata.ts
+++ b/apps/web/charts/analyze/org/activity-action-top-repos/metadata.ts
@@ -9,7 +9,7 @@ const generateMetadata: MetadataGenerator<{
   activity: string;
 }> = ({ parameters: { owner_id, activity }, getOrg }) => {
   const main = getOrg(parseInt(owner_id));
-  const title = getTitle(activity, main.login);
+  const title = getTitle(activity, main?.login);
   return {
     title,
   };

--- a/apps/web/charts/analyze/org/activity-efficiency/visualization.ts
+++ b/apps/web/charts/analyze/org/activity-efficiency/visualization.ts
@@ -190,7 +190,7 @@ export default function (
         type: 'line',
       },
       formatter: (params) => {
-        const { value = {} } = params[0];
+        const { value = {} } = (params as any[])[0];
         const { date, closed, open, merged } = value;
         const parsedDate = DateTime.fromJSDate(new Date(date)).toFormat(
           'yyyy-MM-dd'

--- a/apps/web/charts/analyze/org/activity-map/metadata.ts
+++ b/apps/web/charts/analyze/org/activity-map/metadata.ts
@@ -6,7 +6,7 @@ const generateMetadata: MetadataGenerator<{
 }> = ({ parameters: { owner_id, activity }, getOrg }) => {
   const main = getOrg(owner_id);
   return {
-    title: `Geographical Distribution of ${main.login}`,
+    title: `Geographical Distribution of ${main?.login}`,
   };
 };
 

--- a/apps/web/charts/analyze/org/activity-map/visualization.ts
+++ b/apps/web/charts/analyze/org/activity-map/visualization.ts
@@ -41,7 +41,7 @@ const dataPointToLocationData = (
 ): LocationData[] => {
   return data.map((item) => ({
     country_or_area: item.country_code,
-    count: item[activity],
+    count: (item as any)[activity],
   }));
 };
 
@@ -85,7 +85,7 @@ export default function (
 
   const max = input
     .flat()
-    .reduce((prev, current) => Math.max(prev, current[activity] || 0), 1);
+    .reduce((prev, current) => Math.max(prev, (current as any)[activity] || 0), 1);
 
   const option = {
     dataset: compare(input, (data, name) =>

--- a/apps/web/charts/analyze/org/activity-open-to-close/metadata.ts
+++ b/apps/web/charts/analyze/org/activity-open-to-close/metadata.ts
@@ -3,7 +3,7 @@ import {
   WidgetVisualizerContext,
 } from '@/lib/charts-types';
 
-const generateMetadata: MetadataGenerator<{ owner_id: string; activity }> = ({
+const generateMetadata: MetadataGenerator<{ owner_id: string; activity: string }> = ({
   parameters: { owner_id, activity },
   getOrg,
 }) => {

--- a/apps/web/charts/analyze/org/activity-open-to-close/visualization.ts
+++ b/apps/web/charts/analyze/org/activity-open-to-close/visualization.ts
@@ -90,7 +90,7 @@ export default function (
       axisLabel: { formatter: fmtHours },
       axisPointer: {
         label: {
-          formatter: ({ value }) => fmtHours(Number(value)),
+          formatter: ({ value }: { value: any }) => fmtHours(Number(value)),
         },
       },
       max,
@@ -114,7 +114,7 @@ export default function (
     tooltip: axisTooltip('cross', {
       renderMode: 'html',
       formatter: (params) => {
-        const { value, marker } = params[0];
+        const { value, marker } = (params as any[])[0];
         return `
         ${marker as string}
         <span>${value.repo_name}</span>

--- a/apps/web/charts/analyze/org/activity-open-to-first-response/visualization.ts
+++ b/apps/web/charts/analyze/org/activity-open-to-first-response/visualization.ts
@@ -90,7 +90,7 @@ export default function (
       axisLabel: { formatter: fmtHours },
       axisPointer: {
         label: {
-          formatter: ({ value }) => fmtHours(Number(value)),
+          formatter: ({ value }: { value: any }) => fmtHours(Number(value)),
         },
       },
       max,
@@ -113,7 +113,7 @@ export default function (
     tooltip: axisTooltip('cross', {
       renderMode: 'html',
       formatter: (params) => {
-        const { value, marker } = params[0];
+        const { value, marker } = (params as any[])[0];
         return `
         ${marker as string}
         <span>${value.repo_name}</span>

--- a/apps/web/charts/analyze/org/company/metadata.ts
+++ b/apps/web/charts/analyze/org/company/metadata.ts
@@ -9,7 +9,7 @@ const generateMetadata: MetadataGenerator<{
 }> = ({ parameters: { owner_id, activity }, getOrg }) => {
   const main = getOrg(parseInt(owner_id));
   return {
-    title: `Company Affiliation of ${main.login}`,
+    title: `Company Affiliation of ${main?.login}`,
   };
 };
 

--- a/apps/web/charts/analyze/org/company/visualization.tsx
+++ b/apps/web/charts/analyze/org/company/visualization.tsx
@@ -1,5 +1,6 @@
 import type { WidgetVisualizerContext } from '@/lib/charts-types';
 
+// @ts-expect-error d3-cloud has no type declarations
 import cloud from 'd3-cloud';
 
 type Params = {
@@ -31,7 +32,7 @@ function transformCompanyData (
   }
   return data.flatMap((item, index) => ({
     text: esc(item.organization_name),
-    size: item[valueIndex],
+    size: (item as any)[valueIndex],
     color: undefined,
   }));
 }
@@ -89,9 +90,9 @@ export default async function (
       .padding(2)
       .rotate(0)
       .font('Heiti TC')
-      .fontSize(function (d) { return d.size; })
+      .fontSize(function (d: any) { return d.size; })
       .random(() => 0.5)
-      .on('end', (word) => resolve(word));
+      .on('end', (word: any) => resolve(word));
     layout.start();
     if (signal) {
       signal.onabort = () => {

--- a/apps/web/charts/analyze/org/engagement-scatter/metadata.ts
+++ b/apps/web/charts/analyze/org/engagement-scatter/metadata.ts
@@ -6,7 +6,7 @@ const generateMetadata: MetadataGenerator<{ owner_id: number }> = ({
 }) => {
   const main = getOrg(owner_id);
   return {
-    title: `Most engaged people of ${main.login}`,
+    title: `Most engaged people of ${main?.login}`,
   };
 };
 

--- a/apps/web/charts/analyze/org/pull-requests-open-to-review/visualization.ts
+++ b/apps/web/charts/analyze/org/pull-requests-open-to-review/visualization.ts
@@ -90,7 +90,7 @@ export default function (
       axisLabel: { formatter: fmtHours },
       axisPointer: {
         label: {
-          formatter: ({ value }) => fmtHours(Number(value)),
+          formatter: ({ value }: { value: any }) => fmtHours(Number(value)),
         },
       },
       max,
@@ -114,7 +114,7 @@ export default function (
     tooltip: axisTooltip('cross', {
       renderMode: 'html',
       formatter: (params) => {
-        const { value, marker } = params[0];
+        const { value, marker } = (params as any[])[0];
         return `
         ${marker as string}
         <span>${value.repo_name}</span>

--- a/apps/web/charts/analyze/org/recent-stats/metadata.ts
+++ b/apps/web/charts/analyze/org/recent-stats/metadata.ts
@@ -10,7 +10,7 @@ const generateMetadata: MetadataGenerator<{
 }> = ({ parameters: { owner_id, activity }, getOrg }) => {
   const main = getOrg(parseInt(owner_id));
   return {
-    title: `${upperFirst(activity)} trends of ${main.login}`,
+    title: `${upperFirst(activity)} trends of ${main?.login}`,
   };
 };
 

--- a/apps/web/charts/compose/org/activity-active-ranking/visualization.tsx
+++ b/apps/web/charts/compose/org/activity-active-ranking/visualization.tsx
@@ -93,10 +93,10 @@ const handleTotal = (total: TotalDataPoint[] | undefined) => {
   }
   const { current_period_total, past_period_total } = total?.[0] || {};
 
-  const currentSum = current_period_total;
-  const pastSum = past_period_total;
+  const currentSum = current_period_total ?? 0;
+  const pastSum = past_period_total ?? 0;
   const diff = currentSum - pastSum;
-  const diffPercentage = number2percent(diff / pastSum);
+  const diffPercentage = number2percent(diff / (pastSum || 1));
   return {
     current_period_total,
     past_period_total,

--- a/apps/web/charts/compose/org/activity-growth-total/visualization.tsx
+++ b/apps/web/charts/compose/org/activity-growth-total/visualization.tsx
@@ -85,10 +85,10 @@ const handleTotal = (total: TotalDataPoint[] | undefined) => {
   const { current_period_total, past_period_total } =
     total?.[0] || {};
 
-  const currentSum = current_period_total;
-  const pastSum = past_period_total;
+  const currentSum = current_period_total ?? 0;
+  const pastSum = past_period_total ?? 0;
   const diff = currentSum - pastSum;
-  const diffPercentage = number2percent(diff/pastSum);
+  const diffPercentage = number2percent(diff / (pastSum || 1));
   return {
     current_period_total,
     past_period_total,

--- a/apps/web/charts/compose/org/overview-stars/visualization.tsx
+++ b/apps/web/charts/compose/org/overview-stars/visualization.tsx
@@ -45,10 +45,10 @@ export default function Card({ data: input, ctx, linkedData }: { data: any[]; ct
 
   const { current_period_total, past_period_total } = total?.[0] ?? {};
 
-  const currentStarsSum = current_period_total;
-  const pastStarsSum = past_period_total;
+  const currentStarsSum = current_period_total ?? 0;
+  const pastStarsSum = past_period_total ?? 0;
   const diff = currentStarsSum - pastStarsSum;
-  const diffPercentage = number2percent(diff / pastStarsSum);
+  const diffPercentage = number2percent(diff / (pastStarsSum || 1));
 
   const stars = transferData2Star(data);
 

--- a/apps/web/charts/compose/org/overview-stats/visualization.tsx
+++ b/apps/web/charts/compose/org/overview-stats/visualization.tsx
@@ -59,10 +59,10 @@ export default function Card({ data: input, ctx, linkedData }: { data: any[]; ct
 
   const { current_period_total, past_period_total } = total?.[0] ?? {};
 
-  const currentSum = current_period_total;
-  const pastSum = past_period_total;
+  const currentSum = current_period_total ?? 0;
+  const pastSum = past_period_total ?? 0;
   const diff = currentSum - pastSum;
-  const diffPercentage = number2percent(diff / pastSum);
+  const diffPercentage = number2percent(diff / (pastSum || 1));
 
   const stars = transferData2Star(data);
 

--- a/apps/web/charts/compose/org/participants-growth/visualization.tsx
+++ b/apps/web/charts/compose/org/participants-growth/visualization.tsx
@@ -32,10 +32,10 @@ const handleTotal = (total: TotalDataPoint[] | undefined) => {
   }
   const { current_period_total, past_period_total } = total?.[0] || {};
 
-  const currentSum = current_period_total;
-  const pastSum = past_period_total;
+  const currentSum = current_period_total ?? 0;
+  const pastSum = past_period_total ?? 0;
   const diff = currentSum - pastSum;
-  const diffPercentage = number2percent(diff / pastSum);
+  const diffPercentage = number2percent(diff / (pastSum || 1));
   return {
     current_period_total,
     past_period_total,

